src/ngx_http_lua_common.h

```diff
@@ -176,6 +178,9 @@ struct ngx_http_lua_main_conf_s {
 
     ngx_array_t         *preload_hooks; /* of ngx_http_lua_preload_hook_t */
 
+    ngx_array_t         *delay_list; /* of void* */
+    unsigned             delay_list_inited;
```

Comment:
@agentzh I have use (`ngx_http_lua_delay_init_declare`)[https://github.com/openresty/lua-nginx-module/pull/1050/files#diff-3ed27dd07994dffb9efa8c835e198a15R243] to check that every third part module has register or not. 
If we only use a number as counter instead of array `delay_list`, we have to ensure the third part module register only not more than twice.